homeassistant/components/ialarmxr/config_flow.py

```diff
@@ -0,0 +1,75 @@
+"""Config flow for Antifurto365 iAlarmXR integration."""
+from __future__ import annotations
+
+import logging
+from logging import Logger
+from typing import Any
+
+from pyialarmxr import IAlarmXR, IAlarmXRGenericException
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+_LOGGER: Logger = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST, default=IAlarmXR.IALARM_P2P_DEFAULT_HOST): str,
+        vol.Required(CONF_PORT, default=IAlarmXR.IALARM_P2P_DEFAULT_PORT): int,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+async def _get_device_mac(
+    hass: core.HomeAssistant, username: str, password: str, host: str, port: int
+) -> str:
+    ialarmxr = IAlarmXR(username, password, host, port)
+    return await hass.async_add_executor_job(ialarmxr.get_mac)
```

Comment:
Usually we normalize mac addresses with the `format_mac` helper

Reply:
I suggest moving `await hass.async_add_executor_job(ialarmxr.get_mac)` into a helper function (maybe in a `util.py`) that normalizes the mac and use it from `__init__` as well

Reply:
Uh ok, but mac address will never shown! It's for unique id generation only. Anyway if you suggest me to change it I'll take your suggestion 👍

Reply:
We should normalize it since if you get it from discovery later you can pass everything through `format_mac` and they will all match so you won't have differing formatting for unique ids